backend/apid/routers/namespaces.go

```diff
@@ -30,8 +48,72 @@ func (r *NamespacesRouter) Mount(parent *mux.Router) {
 	}
 
 	routes.Del(r.handlers.DeleteResource)
-	routes.Get(r.handlers.GetResource)
+	routes.Get(r.get)
 	routes.List(r.handlers.ListResources, corev2.NamespaceFields)
 	routes.Post(r.handlers.CreateResource)
 	routes.Put(r.handlers.CreateOrUpdateResource)
 }
+
+func (r *NamespacesRouter) get(req *http.Request) (interface{}, error) {
+	attrs := authorization.GetAttributes(req.Context())
+	if attrs == nil {
+		return nil, actions.NewErrorf(actions.InvalidArgument)
+	}
+
+	resources := []*corev2.Namespace{}
+	err := r.store.ListResources(req.Context(), corev2.NamespacesResource, &resources, &store.SelectionPredicate{})
+	if err != nil {
+		return nil, actions.NewError(actions.InternalErr, err)
+	}
+
+	namespaceMap := make(map[string]*corev2.Namespace, len(resources))
+	for _, namespace := range resources {
+		namespaceMap[namespace.Name] = namespace
+	}
+
+	namespaces := make([]*corev2.Namespace, 0, len(resources))
+
+	var funcErr error
+
+	// Iterate over the rbac rules to discover namespaces that this
+	// user has read access to.
+	r.auth.VisitRulesFor(req.Context(), attrs, func(binding rbac.RoleBinding, rule corev2.Rule, err error) (terminate bool) {
```

Comment:
So this was actually a bug in sensu-enterprise-go; the returned boolean value should actually be named `continue` (it's a reserved keyboard so it needs to be something else similar) and not `terminate`, because if the [return value is `false`, it stops the iteration over the rules](https://github.com/sensu/sensu-go/blob/66ab843069daaca27301b9378f8718f43e47d453/backend/authorization/rbac/rbac.go#L64).

Therefore, we probably need to inverse the return values below

Reply:
This is now fixed and covered by a unit test.